sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2173,6 +2173,37 @@ class Dataset[T] private[sql](
       valueColumnName: String): DataFrame =
     unpivot(ids.toArray, variableColumnName, valueColumnName)
 
+  /**
+   * Transpose a DataFrame, switching rows to columns.
+   * This function transforms the DataFrame such that the values in the specified index
+   * column become the new columns of the DataFrame. If no index column is provided, the first
+   * column is used as the default.
+   *
+   * Please note:
+   * - The values transposed must share the least common type.
+   * - The name of the column into which the original column names are transposed defaults to "key".
+   * - Non-"key" column names for the transposed table are ordered in ascending order.
+   *
+   * @param indexColumn The single column that will be treated as the index for the transpose
+   * operation.This column will be used to pivot the data, transforming the DataFrame such that
+   * the values of the indexColumn become the new columns in the transposed DataFrame. If not
+   * provided, the first column is used.
+   * 
+   * @group untypedrel
+   * @since 4.0.0
+   */
+  def transpose(indexColumn: Option[Column] = None): DataFrame = withPlan {
```

Comment:
It's better to have two overloads, one with indexColumn parameter and one without.

Reply:
Used overloads instead, thanks!